riff-raff/app/assets/javascripts/form-autocomplete.coffee

```diff
@@ -20,6 +21,62 @@ updateDeployInfo = () ->
       $("[rel='tooltip']").tooltip()
   )
 
+addFavourite = (project) ->
+  favourites = JSON.parse(localStorage.getItem('favouriteProjects'))
+  newFavourites =
+    if favourites?
+      favourites.push(project)
+      favourites
+    else
+      [project]
+  localStorage.setItem('favouriteProjects', JSON.stringify(newFavourites))
+  renderFavourites()
+
+deleteFavourite = (project) ->
+  favourites = JSON.parse(localStorage.getItem('favouriteProjects'))
+  newFavourites =
+    if favourites?
+      favourites.filter (fav) -> fav != project
+    else
+      []
+  localStorage.setItem('favouriteProjects', JSON.stringify(newFavourites))
+  renderFavourites()
+
+setupFavouriteHandlers = () ->
+  $('.delete-favourite-project-button').click (e) ->
+    e.preventDefault()
+    selectedProject = e.currentTarget.value
+    if selectedProject?
+      deleteFavourite(selectedProject)
+
+  $('.select-favourite-project-button').click (e) ->
+    e.preventDefault()
+    project = e.target.value
+
+    elemProjectInput = $('#projectInput')
+    elemProjectInput.val(project)
+    updateDeployInfo()
```

Comment:
I had to put all the favourites stuff in this script because it needs to be able to call `updateDeployInfo`, and it doesn't appear to be possible to import functions from other files

Reply:
Not sure how coffeeScript works, but just wanted to mention there’s a nice `@guardian/lib` storage tool: https://github.com/guardian/libs/blob/main/src/storage.README.md